embedded/store/immustore.go

```diff
@@ -2278,17 +2507,48 @@ func (s *ImmuStore) sync() error {
 		return err
 	}
 
-	// will overwrite partially written and uncommitted data
-	err = s.cLog.SetOffset(int64(s.committedTxID * cLogEntrySize))
+	err = s.durablePrecommitWHub.DoneUpto(s.preCommittedTxID)
 	if err != nil {
 		return err
 	}
 
-	_, _, err = s.cLog.Append(s.cLogBuf[:int(s.preCommittedTxID-s.committedTxID)*cLogEntrySize])
+	commitAllowedUpToTxID := s.commitAllowedUpTo()
+	txsCountToBeCommitted := int(commitAllowedUpToTxID - s.committedTxID)
+
+	// will overwrite partially written and uncommitted data
+	err = s.cLog.SetOffset(int64(s.committedTxID * cLogEntrySize))
 	if err != nil {
 		return err
 	}
 
+	var commitUpToTxID uint64
+	var commitUpToTxAlh [sha256.Size]byte
+
+	for i := 0; i < txsCountToBeCommitted; i++ {
+		txID, alh, txOff, txSize, err := s.cLogBuf.readAhead(i)
+		if err != nil {
+			return err
+		}
+
+		var cb [cLogEntrySize]byte
+		binary.BigEndian.PutUint64(cb[:], uint64(txOff))
+		binary.BigEndian.PutUint32(cb[offsetSize:], uint32(txSize))
+
+		_, _, err = s.cLog.Append(cb[:])
+		if err != nil {
+			return err
+		}
+
+		commitUpToTxID = txID
+		commitUpToTxAlh = alh
+	}
+
+	if commitUpToTxID != commitAllowedUpToTxID {
+		// added as a safety fuse but this situation should NOT happen
+		return fmt.Errorf("%w: may commit up to %d but actual transaction to be committed is %d",
```

Comment:
Currently starting the master with one replica `./immudb --replication-sync-followers 1` and it gets stuck at this point in a loop